radar_control/radar_control.py

```diff
@@ -674,7 +674,9 @@ def radar():
                         # we have enough time left to run the integration period.
                         time_elapsed = integration_period_start_time - start_minute
                         if scan_iter < len(scan.scanbound) - 1:
-                            scanbound_time = scan.scanbound[scan_iter+1]
+                            scanbound_time = scan.scanbound[scan_iter + 1]
+                            # TODO: may be negative, perhaps calculate which 'beam' instead
```

Comment:
This comment is confusing, I don't think it's scanbound_time that will be negative (that comes from the slice's scanbound in setup). I think this comment could be reworded?

Also, I think calculating the beam will only be possible with a rework of this code to setup an experiment-wide 'scanbound'. 
```suggestion
                            # TODO: scanbound_time could be in the past if system has taken 
                            # to long, perhaps calculate which 'beam' (scan_iter) instead by 
                            # rewriting this code for an experiment-wide scanbound attribute instead
                            # of individual scanbounds inside the scan objects
```

Reply:
oops, that comment is meant for the line below it, bound_time_remaining, which ends up being negative often (at least, when this issue was happening), which is why I think maybe it's a good idea to skip ahead. 

As an example, given that bound_time_remaining is -8 seconds, and we're running the standard 3.5s intt, that means we should really be two 'beams' in the future, otherwise there will be 0 averages recorded for this iteration and the next before we loop to the appropriate 'beam', where we will then have < 3 seconds for the integration. It  sounds like that's not so easy without a rewrite though.